### PR TITLE
add developer styleguide

### PR DESCRIPTION
**Pull request by tvikito**
| Q             | A
| ------------- | ---
|Description, reason for the PR| add developer styleguide
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://tv-ssp-2185-add-developer-styleguide.odin.shopsys.cloud
  - https://cz.tv-ssp-2185-add-developer-styleguide.odin.shopsys.cloud

### CODE REVIEW

File: project-base/storefront/components/Pages/Styleguide/StyleguideColors.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import React from 'react';
+import { twJoin } from 'tailwind-merge';
+
+class ColorItem {
+    constructor(
+        public name: string,
+        public className: string,
+        // eslint-disable-next-line no-empty-function
+    ) {}
+}
```
Comment (sebaholesz) [inline project-base/storefront/components/Pages/Styleguide/StyleguideColors.tsx:11]: Just a question, why to use classes instead of just regular objects?

**Comment by sebaholesz** [comment]
I see that not all components are used and shown, such as uncontrolled text input, password and search inputs, etc. I am not sure how deep we want to go, just a question to consider

**Comment by sebaholesz** [comment]
My question regarding this styleguide would be if we want to make it dynamic, or that's defeating the purpose (meaning that person viewing it can change some component params, such as disabled, or step for spinbox, etc)? Or if not that, do we want to show snippets of code responsible for that component to explain props? Maybe nobody would use that, its just for you to consider.

File: project-base/storefront/components/Pages/Styleguide/StyleguideColors.tsx (hunk 1 above)
Comment (tvikito) [reply project-base/storefront/components/Pages/Styleguide/StyleguideColors.tsx:11]: Because it easier to create new values

**Review by sebaholesz — APPROVED** [review]

**Pull request merged**